dotty-tensorflow/src/main/scala/me/shadaj/scalapy/tensorflow/api/scalaUtils/PythonUnion.scala

```diff
@@ -6,14 +6,22 @@ import scala.reflect.ClassTag
 
 object PythonUnion {
 
-  def fromPyhtonTypeAndScalaTypeUnion[X <: py.Any, A <: PythonType[X], B <: Any](u: A | B)(implicit ev1: ClassTag[A], ev2: ClassTag[B]): py.|[B, X] =
+  implicit def fromEnumPythonTypeUnion[A <: PythonEnum, X <: py.Any, B <: PythonType[X]](u: A | B)(implicit ev1: ClassTag[A], ev2: ClassTag[B]): py.|[String, X] = {
+    u match {
+      case a: A => py.|.fromLeft(a.v)
+      case b: B => py.|.fromRight(b.underlying)
+      case _    => throw new IllegalArgumentException()
+    }
+  }
+
+  def fromPythonTypeAndScalaTypeUnion[X <: py.Any, A <: PythonType[X], B <: Any](u: A | B)(implicit ev1: ClassTag[A], ev2: ClassTag[B]): py.|[B, X] =
     u match {
       case a: A => py.|.fromRight(a.underlying)
       case b: B => py.|.fromLeft(b)
       case _    => throw new IllegalArgumentException()
     }
 
-  implicit def fromPyhtonTypesUnion[X <: py.Any, Y <: py.Any, A <: PythonType[X], B <: PythonType[Y]](
+  implicit def fromPythonTypesUnion[X <: py.Any, Y <: py.Any, A <: PythonType[X], B <: PythonType[Y]](
```

Comment:
this comes from my mess up in git history it should be ok when we merge our PR stack :DD